From 08c1dd69b556b0325fe6cdfea4b28b4f2df90658 Mon Sep 17 00:00:00 2001
From: Luca Weiss <luca@z3ntu.xyz>
Date: Sat, 28 Jan 2023 17:24:03 +0100
Subject: [PATCH] qcam: Decrease minimum width of selector dialog

On phone screens the default width is too wide, so the OK button cannot
be clicked.

Fix this by decreasing the minimum size of the dialog so it fits nicely.

Signed-off-by: Luca Weiss <luca@z3ntu.xyz>
---
 src/apps/qcam/cam_select_dialog.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/apps/qcam/cam_select_dialog.cpp b/src/apps/qcam/cam_select_dialog.cpp
index 3c8b12a9..2a600383 100644
--- a/src/apps/qcam/cam_select_dialog.cpp
+++ b/src/apps/qcam/cam_select_dialog.cpp
@@ -25,6 +25,9 @@ CameraSelectorDialog::CameraSelectorDialog(libcamera::CameraManager *cameraManag
 	/* Use a QFormLayout for the dialog. */
 	QFormLayout *layout = new QFormLayout(this);
 
+	/* Decrease minimum width of dialog to fit on narrow screens */
+	setMinimumSize(250, 100);
+
 	/* Setup the camera id combo-box. */
 	cameraIdComboBox_ = new QComboBox;
 	for (const auto &cam : cm_->cameras())
-- 
2.39.1

